scormxblock/scormxblock.py

```diff
@@ -103,24 +173,41 @@ def studio_view(self, context=None):
         return frag
 
     def author_view(self, context):
-        html = self.resource_string("static/html/author_view.html")
+        context_html = self.get_context_student()
+        html = self.render_template('static/html/author_view.html', context_html)
         frag = Fragment(html)
         return frag
 
     @XBlock.handler
     def studio_submit(self, request, suffix=''):
+
         self.display_name = request.params['display_name']
         self.has_score = request.params['has_score']
         self.icon_class = 'problem' if self.has_score == 'True' else 'video'
         if hasattr(request.params['file'], 'file'):
             file = request.params['file'].file
             zip_file = zipfile.ZipFile(file, 'r')
-            path_to_file = os.path.join(settings.PROFILE_IMAGE_BACKEND['options']['location'], self.location.block_id)
-            if os.path.exists(path_to_file):
-                shutil.rmtree(path_to_file)
-            zip_file.extractall(path_to_file)
-            self.set_fields_xblock(path_to_file)
-        return Response(json.dumps({'result': 'success'}), content_type='application/json')
+
+            # Create a temporaray directory where the zip will extract.
+            temp_directory = TempFS()
+
+            # Extract the files in the temp directory just created.
+            zip_file.extractall(temp_directory.root_path)
+
+            manifest_path = '{}/imsmanifest.xml'.format(temp_directory.root_path)
+            with open(manifest_path, 'r') as manifest_file:
+                manifest = manifest_file.read()
+            manifest_file.closed
```

Comment:
is this not already closed ?  since there is out of the with statement

Reply:
you're right! I'll change it :)